entity-service/src/test/kotlin/com/egm/stellio/entity/web/EntityHandlerTests.kt

```diff
@@ -1152,20 +1152,84 @@ class EntityHandlerTests {
                     "@context": ["$AQUAC_COMPOUND_CONTEXT"]
                 }
             """.trimIndent()
-        val expectedUpdatedEntity =
+        val attrId = "fishNumber"
+        val updateResult = UpdateResult(
+            updated = arrayListOf(
+                UpdatedDetails(
+                    fishNumberAttribute,
+                    null,
+                    UpdateOperationResult.UPDATED
+                )
+            ),
+            notUpdated = arrayListOf()
+        )
+        every { entityService.exists(any()) } returns true
+        every { authorizationService.userCanUpdateEntity(entityId, "mock-user") } returns true
+        every { entityAttributeService.partialUpdateEntityAttribute(any(), any(), any()) } returns updateResult
+        every { entityService.getFullEntityById(entityId, true) } returns expandJsonLdEntity(deadFish)
+        every { entityEventService.publishPartialUpdateEntityAttributesEvents(any(), any(), any(), any(), any()) } just
+            Runs
+
+        webClient.patch()
+            .uri("/ngsi-ld/v1/entities/$entityId/attrs/$attrId")
+            .header("Link", aquacHeaderLink)
+            .contentType(MediaType.APPLICATION_JSON)
+            .bodyValue(jsonLdFile)
+            .exchange()
+            .expectStatus().isNoContent
+
+        verify { entityService.exists(entityId) }
+        verify {
+            entityAttributeService.partialUpdateEntityAttribute(eq(entityId), any(), eq(listOf(AQUAC_COMPOUND_CONTEXT)))
+        }
+        verify { entityService.getFullEntityById(entityId, true) }
+        verify {
+            entityEventService.publishPartialUpdateEntityAttributesEvents(
+                eq(entityId),
+                any(),
+                eq(updateResult),
+                any(),
+                eq(listOf(AQUAC_COMPOUND_CONTEXT))
+            )
+        }
+
+        confirmVerified(entityService, entityEventService)
+    }
+
+    @Test
+    fun `partial multi attribute update should return a 204 if JSON-LD payload is correct`() {
+        val jsonLdFile = loadSampleData("aquac/fragments/DeadFishes_partialAttributeUpdate.json")
+        val entityId = "urn:ngsi-ld:DeadFishes:019BN".toUri()
+        val deadFish =
             """
                 {
                     "id": "$entityId",
-                    "type": "DeadFishes"
+                    "type": "DeadFishes",
+                    "@context": ["$AQUAC_COMPOUND_CONTEXT"]
                 }
             """.trimIndent()
         val attrId = "fishNumber"
-
+        val updateResult = UpdateResult(
```

Comment:
I don't get how you can get 2 results whereas you update one attribute instance

Reply:
Indeed, it should be fixed, the updateResult should only contain one UpdatedDetails element.